core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -150,100 +151,111 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val (magicValue, timestamp) = getMessageFormatVersionAndTimestamp(groupPartition)
-      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId),
-        timestamp = timestamp, magicValue = magicValue)
-
-      val partitionOpt = replicaManager.getPartition(Topic.GroupMetadataTopicName, groupPartition)
-      partitionOpt.foreach { partition =>
-        val appendPartition = TopicAndPartition(Topic.GroupMetadataTopicName, groupPartition)
-
-        trace("Marking group %s as deleted.".format(group.groupId))
-
-        try {
-          // do not need to require acks since even if the tombstone is lost,
-          // it will be appended again by the new leader
-          partition.appendMessagesToLeader(new ByteBufferMessageSet(config.offsetsTopicCompressionCodec, tombstone))
-        } catch {
-          case t: Throwable =>
-            error("Failed to mark group %s as deleted in %s.".format(group.groupId, appendPartition), t)
-          // ignore and continue
-        }
+      val magicValueAndTimestampOpt = getMessageFormatVersionAndTimestamp(groupPartition)
+      magicValueAndTimestampOpt match {
+        case Some(magicValueAndTimestamp) =>
```

Comment:
You can use `foreach` to do a side-effect in case the value is `Some`. So:

```scala
getMessageFormatVersionAndTimestamp(groupPartition).foreach { magicValueAndTimestamp =>
  ...
}
```